rskj-core/src/main/java/co/rsk/Start.java

```diff
@@ -17,26 +17,32 @@
  */
 package co.rsk;
 
+import co.rsk.cli.RskCli;
 import co.rsk.util.PreflightChecksUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
 
 import javax.annotation.Nonnull;
 
 /**
  * The entrypoint for the RSK full node
  */
 public class Start {
-
     private static final Logger logger = LoggerFactory.getLogger("start");
-
     public static void main(String[] args) {
         setUpThread(Thread.currentThread());
 
+        RskCli rskCli = new RskCli();
```

Comment:
I would also suggest the command line operatioins within `RskCli` this way:

Inside RskCli
```
class RskCli ... {
 
public RskCli(String[] args) {
  load(args);
}

private void load(String[] args) {
  CommandLine commandLine = new CommandLine(this);
  int exitCode = commandLine.execute(args);
  
  if (exitCode != 0 || commandLine.isVersionHelpRequested() || commandLine.isUsageHelpRequested()) {
            throw YourExceptionHere...
  }
}

.
.
.
}
```

Then you create use it like this:

```
RskCli rskCli = new RskCli(args);
.
.
.
ctx = new RskContext(rskCli);
```

Reply:
I've tried a slightly different approach, but the core idea remains the same. the code is much cleaner now. currently fixing some unit tests... once I'm done, I'll push the changes so you can check them. thanks :D